vars/runGradleCheck.groovy

```diff
@@ -40,6 +40,14 @@ void call(Map args = [:]) {
                 rm -rf search
                 git clone ${git_repo_url} search
                 cd search/
+                
+                if \$((git rev-parse -q --verify \"${git_reference}^{commit}\" > /dev/null 2>&1); then
```

Comment:
This exist seems way too late.
Probably better check during the part of github actions instead.
Thoughts?

Reply:
Adding it in github actions will have little to no value, as the user can force push just after the check is done in actions.  The probability and time gap is way more than adding this check just after the head repo is checked out for final processing.

Reply:
With this on Jenkins tho we probably waste a runner.

Probably better to have additional stage before the actual run, and use a alpine docker container to check.

Reply:
I'm okay with that if the runners are not re-usable. In that case no change in the library is needed, just write all the login a different stage. 
If the runner is re-usable, then it will not make much of a difference as the same can be used to run a new job one the current one aborts.